multiraft/multiraft.go

```diff
@@ -180,6 +181,55 @@ func (m *MultiRaft) sendEvent(event interface{}) {
 	}
 }
 
+// fanoutHeartbeat sends the given heartbeat to all groups which believe that
+// their leader resides on the sending node.
+func (s *state) fanoutHeartbeat(req *RaftMessageRequest) {
```

Comment:
I looked into combining this with fanoutHeartbeatRequest, but in the end I liked it better as is.